### ISSUE TITLE
[BUG] Linux / Windows 2.0.3 not working

### ISSUE THREAD

**Issue by krbvroc1** [rdiff-backup/rdiff-backup#396]
I 'accidentally' updated all my Linux/Centos servers from rdiff-backup 1 to 2.0.3 (via EPEL). And now my two windows 10 machines can no longer backup to the Linux/Centos servers. After realizing the version mismatch I also updated Windows to 2.0.3, but no success.

Note that my configuration under version 1 was a known working one for many many years.

```
E:\VSSRdiffBackup>rdiff-backup.exe -v9 --test-server --remote-schema "plink.exe -P 22 -ssh -batch -i ".ssh\putty.ppk" %s rdiff-backup --server" "Z:\\\\testing" backup@backup-server::"/testing"
2020-06-08 19:45:59.580182 -0400  <CLIENT-7108>  Using rdiff-backup version 2.0.3
2020-06-08 19:45:59.580182 -0400  <CLIENT-7108>         with cpython E:\VSSRdiffBackup\rdiff-backup.exe version 3.7.5
2020-06-08 19:45:59.580182 -0400  <CLIENT-7108>         on Windows-10-10.0.18362, fs encoding utf-8
2020-06-08 19:45:59.580182 -0400  <CLIENT-7108>  Executing plink.exe -P 22 -ssh -batch -i .ssh\putty.ppk backup@backup rdiff-backup --server
2020-06-08 19:45:59.580182 -0400  <CLIENT-7108>  Client sending (0): ConnectionRequest: Globals.get with 1 arguments
2020-06-08 19:45:59.580182 -0400  <CLIENT-7108>  Client sending (0): version
2020-06-08 19:46:00.235600 -0400  <CLIENT-7108>  Client received (0): 2.0.3
2020-06-08 19:46:00.235600 -0400  <CLIENT-7108>  Registering connection 1
2020-06-08 19:46:00.251268 -0400  <CLIENT-7108>  Client sending (1): ConnectionRequest: SetConnections.init_connection_remote with 1 arguments
2020-06-08 19:46:00.251268 -0400  <CLIENT-7108>  Client sending (1): 1
2020-06-08 19:46:00.251268 -0400  <CLIENT-7108>  Client received (1): None
2020-06-08 19:46:00.251268 -0400  <CLIENT-7108>  Client sending (2): ConnectionRequest: log.Log.setverbosity with 1 arguments
2020-06-08 19:46:00.251268 -0400  <CLIENT-7108>  Client sending (2): 9
2020-06-08 19:46:00.282517 -0400  <CLIENT-7108>  Client received (2): None
2020-06-08 19:46:00.282517 -0400  <CLIENT-7108>  Client sending (3): ConnectionRequest: log.Log.setterm_verbosity with 1 arguments
2020-06-08 19:46:00.298108 -0400  <CLIENT-7108>  Client sending (3): 9
2020-06-08 19:46:00.335935 -0400  <CLIENT-7108>  Client received (3): None
2020-06-08 19:46:00.335935 -0400  <CLIENT-7108>  Client sending (4): ConnectionRequest: rpath.make_file_dict with 1 arguments
2020-06-08 19:46:00.335935 -0400  <CLIENT-7108>  Client sending (4): b'/testing'
2020-06-08 19:46:00.382853 -0400  <CLIENT-7108>  Client received (4): {'type': None}
2020-06-08 19:46:00.382853 -0400  <CLIENT-7108>  Client sending (5): ConnectionRequest: rpath.make_file_dict with 1 arguments
2020-06-08 19:46:00.382853 -0400  <CLIENT-7108>  Client sending (5): b'/testing'
2020-06-08 19:46:00.429727 -0400  <CLIENT-7108>  Client received (5): {'type': None}
2020-06-08 19:46:00.429727 -0400  <CLIENT-7108>  Client sending (6): ConnectionRequest: Time.setcurtime_local with 1 arguments
2020-06-08 19:46:00.429727 -0400  <CLIENT-7108>  Client sending (6): 1591659960
2020-06-08 19:46:00.467541 -0400  <CLIENT-7108>  Client received (6): None
2020-06-08 19:46:00.467541 -0400  <CLIENT-7108>  Client sending (7): ConnectionRequest: Globals.set with 2 arguments
2020-06-08 19:46:00.467541 -0400  <CLIENT-7108>  Client sending (7): client_conn
2020-06-08 19:46:00.467541 -0400  <CLIENT-7108>  Client sending (7): LocalConnection
2020-06-08 19:46:00.514416 -0400  <CLIENT-7108>  Client received (7): None
2020-06-08 19:46:00.514416 -0400  <CLIENT-7108>  Client sending (8): ConnectionRequest: Globals.postset_regexp_local with 3 arguments
2020-06-08 19:46:00.514416 -0400  <CLIENT-7108>  Client sending (8): no_compression_regexp
2020-06-08 19:46:00.514416 -0400  <CLIENT-7108>  Client sending (8): b'(?i).*\\.(gz|z|bz|bz2|tgz|zip|zst|rpm|deb|jpg|jpeg|gif|png|jp2|mp3|mp4|ogg|ogv|oga|ogm|avi|wmv|mpeg|mpg|rm|mov|mkv|flac|shn|pgp|gpg|rz|lz4|lzh|lzo|zoo|lharc|rar|arj|asc|vob|mdf)$'
2020-06-08 19:46:00.514416 -0400  <CLIENT-7108>  Client sending (8): None
2020-06-08 19:46:00.552209 -0400  <CLIENT-7108>  Client received (8): None
2020-06-08 19:46:00.552209 -0400  <CLIENT-7108>  Client sending (9): ConnectionRequest: robust.install_signal_handlers with 0 arguments
2020-06-08 19:46:00.567835 -0400  <CLIENT-7108>  Client received (9): None
2020-06-08 19:46:00.567835 -0400  <CLIENT-7108>  Client sending (10): ConnectionRequest: Hardlink.initialize_dictionaries with 0 arguments
2020-06-08 19:46:00.567835 -0400  <CLIENT-7108>  Client received (10): None
2020-06-08 19:46:00.567835 -0400  <CLIENT-7108>  Exception '' raised of class '<class 'AssertionError'>':
  File "rdiff_backup\Main.py", line 393, in error_check_Main
  File "rdiff_backup\Main.py", line 415, in Main
  File "rdiff_backup\Main.py", line 373, in take_action
  File "rdiff_backup\SetConnections.py", line 301, in TestConnections

Traceback (most recent call last):
  File "rdiff-backup", line 32, in <module>
  File "rdiff_backup\Main.py", line 393, in error_check_Main
  File "rdiff_backup\Main.py", line 415, in Main
  File "rdiff_backup\Main.py", line 373, in take_action
  File "rdiff_backup\SetConnections.py", line 301, in TestConnections
AssertionError
[7108] Failed to execute script rdiff-backup
2020-06-08 19:46:00.296478 -0400  <SERVER-12483>  Server sending (2): None
2020-06-08 19:46:00.301012 -0400  <SERVER-12483>  Server received (3): ConnectionRequest: log.Log.setterm_verbosity with 1 arguments
2020-06-08 19:46:00.341386 -0400  <SERVER-12483>  Server received (3): 9
2020-06-08 19:46:00.341494 -0400  <SERVER-12483>  Server sending (3): None
2020-06-08 19:46:00.345888 -0400  <SERVER-12483>  Server received (4): ConnectionRequest: rpath.make_file_dict with 1 arguments
2020-06-08 19:46:00.386417 -0400  <SERVER-12483>  Server received (4): b'testing'
2020-06-08 19:46:00.386594 -0400  <SERVER-12483>  Server sending (4): {'type': None}
2020-06-08 19:46:00.391291 -0400  <SERVER-12483>  Server received (5): ConnectionRequest: rpath.make_file_dict with 1 arguments
2020-06-08 19:46:00.431392 -0400  <SERVER-12483>  Server received (5): b'testing'
2020-06-08 19:46:00.431524 -0400  <SERVER-12483>  Server sending (5): {'type': None}
2020-06-08 19:46:00.435957 -0400  <SERVER-12483>  Server received (6): ConnectionRequest: Time.setcurtime_local with 1 arguments
2020-06-08 19:46:00.476307 -0400  <SERVER-12483>  Server received (6): 1591659960
2020-06-08 19:46:00.476454 -0400  <SERVER-12483>  Server sending (6): None
2020-06-08 19:46:00.479219 -0400  <SERVER-12483>  Server received (7): ConnectionRequest: Globals.set with 2 arguments
2020-06-08 19:46:00.519311 -0400  <SERVER-12483>  Server received (7): client_conn
2020-06-08 19:46:00.519389 -0400  <SERVER-12483>  Server received (7): PipeConnection 0
2020-06-08 19:46:00.519500 -0400  <SERVER-12483>  Server sending (7): None
2020-06-08 19:46:00.525580 -0400  <SERVER-12483>  Server received (8): ConnectionRequest: Globals.postset_regexp_local with 3 arguments
2020-06-08 19:46:00.565413 -0400  <SERVER-12483>  Server received (8): no_compression_regexp
2020-06-08 19:46:00.565513 -0400  <SERVER-12483>  Server received (8): b'(?i).*\\.(gz|z|bz|bz2|tgz|zip|zst|rpm|deb|jpg|jpeg|gif|png|jp2|mp3|mp4|ogg|ogv|oga|ogm|avi|wmv|mpeg|mpg|rm|mov|mkv|flac|shn|pgp|gpg|rz|lz4|lzh|lzo|zoo|lharc|rar|arj|asc|vob|mdf)$'
2020-06-08 19:46:00.565574 -0400  <SERVER-12483>  Server received (8): None
2020-06-08 19:46:00.565652 -0400  <SERVER-12483>  Server sending (8): None
2020-06-08 19:46:00.569975 -0400  <SERVER-12483>  Server received (9): ConnectionRequest: robust.install_signal_handlers with 0 arguments
2020-06-08 19:46:00.570185 -0400  <SERVER-12483>  Server sending (9): None
2020-06-08 19:46:00.572908 -0400  <SERVER-12483>  Server received (10): ConnectionRequest: Hardlink.initialize_dictionaries with 0 arguments
2020-06-08 19:46:00.573037 -0400  <SERVER-12483>  Server sending (10): None
2020-06-08 19:46:00.605172 -0400  <SERVER-12483>  Exception 'Truncated header string (problem probably originated remotely)' raised of class '<class 'rdiff_backup.connection.ConnectionReadError'>':
  File "/usr/lib64/python3.6/site-packages/rdiff_backup/Main.py", line 393, in error_check_Main
    Main(arglist)
  File "/usr/lib64/python3.6/site-packages/rdiff_backup/Main.py", line 415, in Main
    take_action(rps)
  File "/usr/lib64/python3.6/site-packages/rdiff_backup/Main.py", line 348, in take_action
    connection.PipeConnection(sys.stdin.buffer, sys.stdout.buffer).Server()
  File "/usr/lib64/python3.6/site-packages/rdiff_backup/connection.py", line 422, in Server
    self.get_response(-1)
  File "/usr/lib64/python3.6/site-packages/rdiff_backup/connection.py", line 379, in get_response
    req_num, object = self._get()
  File "/usr/lib64/python3.6/site-packages/rdiff_backup/connection.py", line 284, in _get
    raise ConnectionReadError("Truncated header string (problem "

2020-06-08 19:46:00.605228 -0400  <SERVER-12483>  Fatal Error: Lost connection to the remote system
```

I tried using the built-in Win10 SSH as mentioned in #393 and encountered new issues (same as @bng919 reported ) so I was trying to stick with plink.exe (putty) since it was a known working configuration.

**Comment by ericzolf** [comment]
I admit that the handling isn't properly done, but your call is wrong: you need to call with only the destination server::folder you want to test i.e. in your case without the source parameter `"Z:\\\\testing"` (or without `--test-server` of course, if you wanted to do a backup). As a side note, you can test multiple servers at once but all the parameters must have the form server::folder.

**Comment by krbvroc1** [comment]
> I admit that the handling isn't properly done, but your call is wrong

Wow... since things 'worked' for years with the old rdiff 1.x.x, I never suspected that was the issue. Thanks! After removing that local path the test-server works. 

I had gone down the rabbit hole of messing with ssh versus putty (plink) and there were lots of other issues (and exception messages), so I lost sight of the incorrect args to test-server.

**Comment by ericzolf** [comment]
Then I consider the issue to be solved, it'll get closed once the PR #397 is merged.
